src/main/scala/SbtGithubReleasePlugin.scala

```diff
@@ -19,7 +19,8 @@ case object GithubRelease {
       lazy val commitish = settingKey[String]("Specifies the commitish value that determines where the Git tag is created from")
       lazy val draft = settingKey[Boolean]("true to create a draft (unpublished) release, false to create a published one")
       lazy val prerelease = settingKey[Boolean]("true to identify the release as a prerelease. false to identify the release as a full release")
-      lazy val releaseAssets = taskKey[Seq[File]]("The files to upload")
+      lazy val releaseAssets = taskKey[Map[Artifact, File]]("The artifact files to upload")
+      lazy val artifact2mime = taskKey[(Artifact, File)=> String]("Function mapping an artifact/file to a mime type")
```

Comment:
I don't see much point in adding this as a setting. It can be just a convenience function provided through `autoImport`. If anybody wants to use some other way of retrieving this type, they can just use it directly.